code/controllers/subsystem/mapping.dm

```diff
@@ -83,21 +83,32 @@ var/datum/subsystem/mapping/SSmapping
 
 	seedRuins(space_zlevels, rand(8,16), /area/space, space_ruins_templates)*/
 
-	load_star(SSstarmap.current_system, 1)
-
 	// Set up Z-level transistions.
 	space_manager.do_transition_setup()
 	..()
 
-/datum/subsystem/mapping/proc/load_star(datum/star_system/star, var/is_initial = 0)
+/datum/subsystem/mapping/proc/clear_navbeacon()
+	var/area/spacearea = locate(/area/space)
+	for(var/datum/sub_turf_block/STB in split_block(locate(1, 1, 1), locate(255, 255, 1)))
+		for(var/turf/T in STB.return_list())
+			for(var/A in T.contents)
+				qdel(A) // Clear everything out, not including docking ports
+			for(var/A in T.contents)
```

Comment:
Wouldn't it be better to check for contents of the turf here, then if something's there, recurse the qdel?